griptape/tools/query_client/tool.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import annotations
+
+from attrs import Factory, define, field
+from schema import Literal, Or, Schema
+
+from griptape.artifacts import BaseArtifact, ErrorArtifact, ListArtifact, TextArtifact
+from griptape.config import config
+from griptape.engines.rag import RagEngine
+from griptape.engines.rag.modules import (
+    PromptResponseRagModule,
+)
+from griptape.engines.rag.rag_context import RagContext
+from griptape.engines.rag.stages import ResponseRagStage
+from griptape.mixins.rule_mixin import RuleMixin
+from griptape.tools.rag_client.tool import RagClient
+from griptape.utils.decorators import activity
+
+
+@define(kw_only=True)
+class QueryClient(RagClient, RuleMixin):
```

Comment:
Struggled with this Tool, should this behavior live in `RagClient` or be separate like this?

Reply:
With the [naming refactor](https://github.com/griptape-ai/griptape/pull/1060) we already have a `RagTool`. This Tool subclasses `RagTool` to provide a preconfigured `RagEngine` used for querying an Artifact.